app/services/users/omniauth.rb

```diff
@@ -6,31 +6,37 @@ def initialize(auth)
   end
 
   def get
-    user
+    check_auth
+    @user
   end
 
   private
 
   delegate :provider, :uid, :info, to: :auth
   delegate :email, to: :info
 
-  def user
-    @_user ||= find_user || create_user
+  def find_authorization(user)
+    user.authorizations.where(provider: provider, uid: uid).first
   end
 
-  def find_user
-    User.where(provider: provider, uid: uid, email: email).first
+  def create_authorization(user)
```

Comment:
@sinpai we can avoid passing `user` everwhere as an argument, what do you think about having method `#user`?
```ruby
def user
  @user ||= User.where(email: email).first || create_user
end
```

?